content/blog/analysis-with-RCrunch.Rmd

```diff
@@ -0,0 +1,300 @@
+```{r}
+crtabs(mean(JobSatisfaction)~HomeRemote, so_data)
+```
+
+
+## Other data manipulation
+There are a bunch of repetitive cleaning tasks that Crunch makes easy. First, let's add variable identifiers for text and numeric variables. With a few lines of R code we are able to change many variable names at the same time.
+```{r}
+text_vars <- variables(so_data)[types(variables(so_data)) == "text"]
+names(text_vars) <- paste(names(text_vars), "- text")
+
+numeric_vars <- variables(so_data)[types(variables(so_data)) == "numeric"]
+names(numeric_vars) <- paste(names(numeric_vars), "- numeric")
+```
+
+Another common task is to derive a new variable based on the contents or one or more other variables. With `makeCaseVaraible`, all we have to do is provide a sequence of cases, each with a logical expression that can span multiple variables. Say we wanted to classify professional and student contributors to open source software separately from those who don't contribute at all. Using `makeCaseVariable` saves us from having to download all of the data for both of these columns when all we really need to do is reference the category labels.
+
+```{r}
+case_statements <- list(
+    list(expression=so_data$Professional == "Professional developer" &
+             so_data$ProgramHobby %in% c("Yes, I contribute to open source projects", "Yes, both"),
+         name="Professional open source contributor"),
+    list(expression=so_data$Professional == "Student" &
+             so_data$ProgramHobby %in% c("Yes, I contribute to open source projects", "Yes, both"),
+         name="Student open source contributor"),
+    list(expression=so_data$ProgramHobby %in% c("No", "Yes, I program as a hobby"),
+         name="Open source non-contributor"))
+
+so_data[['OScontributor']] <- makeCaseVariable(cases=case_statements,
+    name = "Open source contribution (professional/student)")
+
+so_data[['OScontributor']]
+```
+
+## `as.data.frame` is helpful for connecting with functions that want a dataframe
+To get an even more data-frame like object, `as.data.frame` will produce a _Crunch dataframe_ object which acts more like a standard R dataframe, but still doesn't pull all of the data from the Crunch server. This is useful for functions that require a dataframe `data` argument (like `lm`). The special part of _Crunch dataframes_ is that although they act like regular dataframes (e.g. you get R vectors when you access its columns), the data for each column is only retrieved when it is needed. 
+
+Using `as.data.frame`, we can fit a linear regression with a single predictor of `HomeRemote` predicting `JobSatisfaction`. The syntax for `lm` is exactly the same as it would be to use a local dataframe. But because we are using a _Crunch dataframe_, only the data that `lm` needs is transferred and only when R needs it.
+```{r}
+so_df <- as.data.frame(so_data)
+fit <- lm(JobSatisfaction~HomeRemote, data=so_df)
+summary(fit)
+```
+
+## When all else fails
```

Comment:
I wouldn’t frame it this way — nothing failed. You just had a different need. In fact, the csv exporter allows you to export a compact 'category id' version of the data (or factor level), filtered, subsets of columns, etc